File: pdu/python/nav_msgs/pdu_cdr_conv_OccupancyGrid.py
from .pdu_pytype_OccupancyGrid import OccupancyGrid
from ..pdu_cdr_runtime import CdrReader, CdrWriter

from ..std_msgs.pdu_cdr_conv_Header import *
from ..nav_msgs.pdu_cdr_conv_MapMetaData import *
from ..geometry_msgs.pdu_cdr_conv_Point import *
from ..geometry_msgs.pdu_cdr_conv_Pose import *
from ..geometry_msgs.pdu_cdr_conv_Quaternion import *
from ..builtin_interfaces.pdu_cdr_conv_Time import *



def py_to_cdr_body_OccupancyGrid(writer: CdrWriter, src: OccupancyGrid):
    py_to_cdr_body_Header(writer, src.header)
    py_to_cdr_body_MapMetaData(writer, src.info)
    writer.write_sequence_length(src.data)
    for elem in src.data:
        writer.write_int8(elem)


def cdr_body_to_py_OccupancyGrid(reader: CdrReader, dst: OccupancyGrid):
    cdr_body_to_py_Header(reader, dst.header)
    cdr_body_to_py_MapMetaData(reader, dst.info)
    dst.data = [reader.read_int8() for _ in range(reader.read_uint32())]
    return dst


def py_to_cdr_OccupancyGrid(src: OccupancyGrid) -> bytes:
    writer = CdrWriter()
    writer.write_encapsulation()
    py_to_cdr_body_OccupancyGrid(writer, src)
    return writer.bytes()


def cdr_to_py_OccupancyGrid(cdr_payload) -> OccupancyGrid:
    reader = CdrReader(cdr_payload)
    reader.read_encapsulation()
    dst = OccupancyGrid()
    return cdr_body_to_py_OccupancyGrid(reader, dst)

File: pdu/javascript/nav_msgs/pdu_cdr_conv_OccupancyGrid.js
import { OccupancyGrid } from './pdu_jstype_OccupancyGrid.js';
import { PduCdrWriter, PduCdrReader } from '../pdu_cdr_runtime.js';
import { Header } from '../std_msgs/pdu_jstype_Header.js';
import { PduHeaderConverter } from '../std_msgs/pdu_cdr_conv_Header.js';
import { MapMetaData } from '../nav_msgs/pdu_jstype_MapMetaData.js';
import { PduMapMetaDataConverter } from '../nav_msgs/pdu_cdr_conv_MapMetaData.js';
import { Point } from '../geometry_msgs/pdu_jstype_Point.js';
import { PduPointConverter } from '../geometry_msgs/pdu_cdr_conv_Point.js';
import { Pose } from '../geometry_msgs/pdu_jstype_Pose.js';
import { PduPoseConverter } from '../geometry_msgs/pdu_cdr_conv_Pose.js';
import { Quaternion } from '../geometry_msgs/pdu_jstype_Quaternion.js';
import { PduQuaternionConverter } from '../geometry_msgs/pdu_cdr_conv_Quaternion.js';
import { Time } from '../builtin_interfaces/pdu_jstype_Time.js';
import { PduTimeConverter } from '../builtin_interfaces/pdu_cdr_conv_Time.js';


export class PduOccupancyGridConverter {
    /**
     * @param {PduCdrWriter} writer
     * @param { OccupancyGrid } src
     */
    static to_cdr_body(writer, src) {
        PduHeaderConverter.to_cdr_body(writer, src.header);
        PduMapMetaDataConverter.to_cdr_body(writer, src.info);
        writer.write_sequence_length(src.data);
        for (const elem of src.data) {
            writer.write_int8(elem);
        }
    }

    /**
     * @param {PduCdrReader} reader
     * @param { OccupancyGrid } dst
     * @returns { OccupancyGrid }
     */
    static cdr_body_to_js(reader, dst) {
        PduHeaderConverter.cdr_body_to_js(reader, dst.header);
        PduMapMetaDataConverter.cdr_body_to_js(reader, dst.info);
        dst.data = Array.from({ length: reader.read_uint32() }, () => reader.read_int8());
        return dst;
    }

    /**
     * @param { OccupancyGrid } src
     * @returns {ArrayBuffer}
     */
    static to_cdr(src) {
        const writer = new PduCdrWriter();
        writer.write_encapsulation();
        this.to_cdr_body(writer, src);
        return writer.get_buf();
    }

    /**
     * @param {ArrayBuffer|ArrayBufferView} cdrPayload
     * @returns { OccupancyGrid }
     */
    static from_cdr(cdrPayload) {
        const reader = new PduCdrReader(cdrPayload);
        reader.read_encapsulation();
        return this.cdr_body_to_js(reader, new OccupancyGrid());
    }
}
